pkg/logbridge/logbridge_test.go

```diff
@@ -0,0 +1,180 @@
+package logbridge
+
+import (
+	"bytes"
+	"errors"
+	"testing"
+	"time"
+
+	"github.com/square/p2/pkg/logging"
+)
+
+const newLine = byte(10)
+
+type TrackingWriter struct {
+	numWrites int
+}
+
+func (tw *TrackingWriter) Write(p []byte) (int, error) {
+	tw.numWrites++
+	return len(p), nil
+}
+
+func TestLogBridge(t *testing.T) {
+	type testCase struct {
+		inputSize      int
+		bridgeCapacity int
+		expected       int
+	}
+
+	testCases := []testCase{
+		{inputSize: 0, bridgeCapacity: 0, expected: 0},
+		{inputSize: 10, bridgeCapacity: 10, expected: 10},
+		{inputSize: 100, bridgeCapacity: 10, expected: 10},
+		{inputSize: 10, bridgeCapacity: 100, expected: 10},
+	}
+
+	for i, testCase := range testCases {
+		t.Logf("test case %d", i)
+		input := make([]byte, 0, 2*testCase.inputSize)
+		for i := 0; i < testCase.inputSize; i++ {
+			input = append(input, byte('a'), newLine)
+		}
+
+		reader := bytes.NewReader(input)
+		writer := &TrackingWriter{}
+
+		LossyCopy(writer, reader, testCase.bridgeCapacity, logging.DefaultLogger)
```

Comment:
Doesn't this only verify that the synchronous retried copy worked? Is there any way to verify the `go lossyCopy()` part of the `LossyCopy` function?